policy/compiler/compiler.go

```diff
@@ -401,6 +412,11 @@ func (tc *templateCompiler) compileValidator(dyn *model.DynValue, ctmpl *model.T
 func (tc *templateCompiler) compileValidatorOutputDecisions(
 	prods *model.DynValue, env *cel.Env, ceval *model.Evaluator) {
 	productions := tc.listValue(prods)
+	if len(productions.Entries) > tc.limits.ValidatorProductionLimit {
```

Comment:
Same suggestion here (and below) as the previous ones. Try to report the production where the user trips the limit. Given your test case structure, the offending line should be pretty easy to spot, but this won't always be the case for users.